### PR TITLE
[Doc] Release notes for v8.4.1

### PR DESCRIPTION
**Pull request by ywangd**
Gradle generated release notes for v8.4.1 as part of release process.

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine run elasticsearch-ci/docs

File: docs/reference/release-notes/highlights.asciidoc [hunk 1]
```diff
@@ -17,95 +17,13 @@ Other versions:
 | {ref-bare}/8.1/release-highlights.html[8.1]
 | {ref-bare}/8.0/release-highlights.html[8.0]
 
+// The notable-highlights tag marks entries that
+// should be featured in the Stack Installation and Upgrade Guide:
 // tag::notable-highlights[]
-
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/highlights.asciidoc:21]: I don't think that these release highlights should be removed. The published [highlights for 8.4](https://www.elastic.co/guide/en/elasticsearch/reference/current/release-highlights.html) have several inclusions, which should remain. I think that this entire section needs to stay.

File: docs/reference/migration/migrate_8_4.asciidoc [hunk 2]
```diff
@@ -20,33 +20,3 @@ coming::[8.4.0]
 There are no breaking changes in {es} 8.4.
 // end::notable-breaking-changes[]
 
-
```
Comment (lockewritesdocs) [inline docs/reference/migration/migrate_8_4.asciidoc:23]: I don't think that this information should be removed. The [migration page](https://www.elastic.co/guide/en/elasticsearch/reference/current/migrating-8.4.html) for 8.4 contains deprecation information that relates to the minor version release, and shouldn't be removed for a patch release.

File: docs/reference/migration/migrate_8_4.asciidoc [hunk 3]
```diff
@@ -9,7 +9,7 @@ your application to {es} 8.4.
 
 See also <<release-highlights>> and <<es-release-notes>>.
 
-coming::[8.4.0]
+coming::[8.4.1]
```
Comment (lockewritesdocs) [inline docs/reference/migration/migrate_8_4.asciidoc:12]: We'll need a follow-up PR to remove the `coming` tag when 8.4.1 is GA.

File: docs/reference/release-notes/8.4.1.asciidoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
+[[release-notes-8.4.1]]
+== {es} version 8.4.1
+
+coming[8.4.1]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.4.1.asciidoc:4]: We'll need a follow-up PR to remove the `coming` tag when 8.4.1 is GA.

**Review by lockewritesdocs — CHANGES_REQUESTED** [review]
I don't think that the information in the migration guide or the release highlights should be removed. These pages contain information that applies to the entire minor version, and should not be removed in a patch release.

File: docs/reference/migration/migrate_8_4.asciidoc (hunk 3 above)
Comment (debadair) [reply docs/reference/migration/migrate_8_4.asciidoc:12]: We should just remove the coming tag, it doesn't need to be updated to 8.4.1.

File: docs/reference/migration/migrate_8_4.asciidoc (hunk 2 above)
Comment (debadair) [reply docs/reference/migration/migrate_8_4.asciidoc:23]: Agreed. The info for 8.4 should remain as-is.

File: docs/reference/release-notes/8.4.1.asciidoc (hunk 4 above)
Comment (debadair) [reply docs/reference/release-notes/8.4.1.asciidoc:4]: I can open a PR. We still have some other stray coming tags it looks like.

File: docs/reference/release-notes/highlights.asciidoc (hunk 1 above)
Comment (debadair) [reply docs/reference/release-notes/highlights.asciidoc:21]: Agreed. The info for 8.4 should remain as-is.

**Review by debadair — CHANGES_REQUESTED** [review]

File: docs/reference/release-notes/highlights.asciidoc (hunk 1 above)
Comment (ywangd) [reply docs/reference/release-notes/highlights.asciidoc]: Restored the content.

File: docs/reference/migration/migrate_8_4.asciidoc (hunk 2 above)
Comment (ywangd) [reply docs/reference/migration/migrate_8_4.asciidoc]: Reverted the deletion.

File: docs/reference/migration/migrate_8_4.asciidoc (hunk 3 above)
Comment (ywangd) [reply docs/reference/migration/migrate_8_4.asciidoc]: I removed the `coming` tag.

File: docs/reference/release-notes/8.4.1.asciidoc (hunk 4 above)
Comment (ywangd) [reply docs/reference/release-notes/8.4.1.asciidoc:4]: Thanks @debadair. I'll leave this as is for now.

**Comment by ywangd** [comment]
@debadair @lockewritesdocs Thanks for the review. I restored the deleted contents. They got deleted by running the release note automation gradle task. I was surprised that they got deleted too. But I know little about release process and the automation. So I left it as is. Thank you for catching this issue. I suspect the automation process hasn't really been fine tuned for patch releases. I'll report this back to the delivery team.

**Review by lockewritesdocs — APPROVED** [review]
Changes all LGTM 🦖 Thanks for handling this set of release notes @ywangd!

**Review by debadair — APPROVED** [review]
LGTM

**Pull request merged**